microsoft/devhome
# Modernize and fix the Sample Extension

## Summary of the pull request
This updates the extension sample to be consistent with the latest Dev Home SDK and includes more extensibility examples and conforms better to existing extension paradigms.

## References and relevant issues
#3650 

## Detailed description of the pull request / Additional comments
* Added localization.
* Added logging.
* Added release/dev/canary GUIDs and package manifests.
* Added settings provider with functioning adaptive card with a button to open log location.
* Updated extension to have a DI host and uses it for the Settings provider.
* Changed sample to use latest published SDK as opposed to locally-built SDK
* Removed local SDK from the solution and any references to it.

## Validation steps performed
* Tested extension with locally built dev home and preview dev home, both worked as expected
* Tested extension with locally built SDK (had to update the project to use it) and worked as expected.

## PR checklist
- [x] Closes #3650
